src/pages/groupComparison/OverlapVenn.tsx

```diff
@@ -17,12 +17,15 @@ export interface IVennProps {
     sampleGroups: {
         uid: string;
         cases: string[];
+        color: string;
+        nameWithOrdinal: string;
     }[];
     patientGroups: {
         uid: string;
         cases: string[];
+        color: string;
+        nameWithOrdinal: string;
     }[];
-    uidToGroup: { [uid: string]: ComparisonGroup };
```

Comment:
sorry I should have been clearer - I think it will be a lot more easily maintainable to just keep doing it the way we've been doing - the input is uid, and we just use uidToGroup to get any info that we need. is there a reason you want to change it this way?

Reply:
had to do it this way to maintain group order. and looks like only these two parameters are used from `ComparisonGroup`. do you have any other idea?

Reply:
I dont understand what you can do now re: group order that you couldnt do before?

Reply:
in legendData just loop through sampleGroups as you're doing. that gives the order. then to get color and nameWithOrdinal go through uidToGroup

Reply:
hmm ok